python/lsst/ctrl/bps/drivers.py

```diff
@@ -186,12 +186,20 @@ def acquire_qgraph_driver(config_file, **kwargs):
     qgraph : `lsst.pipe.base.graph.QuantumGraph`
         A graph representing quanta.
     """
-    config = _init_submission_driver(config_file, **kwargs)
-    submit_path = config[".bps_defined.submitPath"]
+    _LOG.info("Initializing execution environment")
+    with time_this(log=_LOG, level=logging.INFO, prefix=None,
+                   msg="Initializing execution environment completed", mem_usage=True, mem_units="GB"):
+        config = _init_submission_driver(config_file, **kwargs)
+        submit_path = config[".bps_defined.submitPath"]
+    _LOG.info("Peak memory usage for bps process %.3f GB (main), %.3f GB (largest child process)",
```

Comment:
I touched the issue briefly in one of my earlier comment, so let me go somewhat more into the details here why I think it may be somewhat confusing for possible users.

As I see it, `time_this` reports numbers which are "local" for the piece of code it was applied to that is its duration and now also the change in the memory usage. However, peak memory usage seem more like a "global" characteristic of the entire execution of the process. As a result peak memory values `time_this` will be reporting may have little to do with the code it is currently timing.

However, if you think that it's not really an issue I will make `time_this` report peak memory usage again.

Reply:
I'm still working on my utils review but I see that `timeMethod` uses the peak values in its reporting. `time_this` may want to report the delta between peak usage at start of context manager and peak usage at end. The current memory  usage after things have run may not be all that interesting other than looking for memory leaks (and measuring it at the start of the context manager and the end may also be interesting).

Reply:
Just for the record, I made `time_this` to report peak delta while working on `lsst.utils`.